xonsh/xontribs.py

```diff
@@ -132,6 +132,11 @@ def xontribs_load(names, verbose=False):
     return res
 
 
+def xontribs_loaded():
+    """Returns list of loaded xontribs."""
+    return [s.split(".")[1] for s in sys.modules if s.startswith("xontrib.")]
```

Comment:
I would rather this not iterate over the entirety of `sys.modules`

maybe something like

```
meta = xontrib_metadata()
for md in meta["xontribs"]:
    [spec.name in sys.modules for spec in find_xontrib(md["name"])
```

Reply:
My key idea is to show all xontribs. Not only that described in xontrib file.

Reply:
I also tried to use `importlib` and `pkgutil` for example:
```
import pkgutil
for importer, modname, ispkg in pkgutil.iter_modules(importlib.util.find_spec("xontrib").submodule_search_locati
ons):
     print("Found submodule %s (is a package: %s)" % (modname, ispkg))
```
But it is not working for xontribs with dash in the name. I.e. `readable-traceback`.

Reply:
Sure, but I think maybe xontribs that want to be found should add themselves to `xontribs_metadata`.  Maybe we should add a registration function for that.

Reply:
> xontribs that want to be found should add themselves

This is somewhat contrary to the goal:

> we want to disambiguate a user report where the version of xonsh doesn't include the xontrib info

If user played with some xontribs from github or try to write his own xontrib and foreget about it and then he breaked the xonsh it should run `xonfig` and remeber about his experiments.

Reply:
Yeah, but I think we want to be fast and stable before being fully complete. This is a piece of debugging code, so it needs to work even when other parts of xonsh/python are falling down. Does that make sense?

Reply:
Also, I think that once modules are loaded once, they are added to the metadata, so that `xontrib list` will show them

Reply:
We just need a simplest way to get the list of loaded xontribs. If `importlib` can't help because of missing the xontribs with dash in the name and `xontribs` module can't help because of using only list of known xontribs, I see the one way is to using `sys.modules` or similar. If you know the more optimal way to avoid `sys.modules` please point it out.

Reply:
@gforsyth there is a way to read files in `xontrib` directory instead of iterating `sys.modules`:
```
[x.name.split('.')[0] for x in Path(importlib.util.find_spec("xontrib").submodule_search_locations[0]).glob('*')
]
```
What do you think?

Reply:
Yeah, I think this is better. If you look at what [list does below](https://github.com/xonsh/xonsh/pull/3894/files#diff-0108dd9b8af7697497623c2f97c08b67536c0d7e4d593b5ce3e9656928854663R156-R161), it basically does a 

```
spec = find_xontrib(name)
loaded = spec.name in sys.modules
```

I would actually recommend splitting out the whole first part of the `_list` function into its own top level function for getting the `data` list.  Maybe `xontrib_data()`?  This function could then be reused in `xonfig`

Reply:
Done